python/pyngraph/function.cpp

```diff
@@ -49,4 +51,24 @@ void regclass_pyngraph_Function(py::module m)
             py::cast(self.get_output_shape(0)).attr("__str__")().cast<std::string>();
         return "<" + class_name + ": '" + self.get_friendly_name() + "' (" + shape + ")>";
     });
+    function.def_static("from_capsule", [](py::object* capsule) {
+        auto* pybind_capsule_ptr = capsule->ptr();
+        auto* capsule_ptr = PyCapsule_GetPointer(pybind_capsule_ptr, CAPSULE_NAME);
+        auto* ngraph_function = static_cast<std::shared_ptr<ngraph::Function>*>(capsule_ptr);
+
+        return *ngraph_function;
+    });
+    function.def_static("to_capsule", [](std::shared_ptr<ngraph::Function>& ngraph_function) {
+        auto* sp_copy = new std::shared_ptr<ngraph::Function>(ngraph_function);
+        auto pybind_capsule = py::capsule(sp_copy, CAPSULE_NAME, [](PyObject* capsule) {
```

Comment:
Please add a few 1-line comments explaining the calls made in this code.

Reply:
I think it would be more readable to assign a name to this lambda, so we know what it does - `cleanup_callback` as in the PyBind docs seems like a good name.